src/OpenTelemetry.Api/Trace/ActivityExtensions.cs

```diff
@@ -29,7 +29,7 @@ public static class ActivityExtensions
     /// <param name="activity">Activity instance.</param>
     /// <param name="status">Activity execution status.</param>
     [MethodImpl(MethodImplOptions.AggressiveInlining)]
-    [Obsolete("Call Activity.SetStatus instead this method will be removed in a future version.")]
+    [Obsolete("Call Activity.SetStatus(ActivityStatusCode code, string? description = null) instead. This method will be removed in a future version.")]
```

Comment:
I didn't spend time looking at what others are doing or what's the official guidance (if there is any). IIRC the typical pattern is to mention the method name rather than the parameters while using ObsoleteAttribute.

Reply:
@reyang, If you prefer, we could use this. I was aware about these changes and I needed some time to find correct method to fix warnings/errors.
```suggestion
    [Obsolete("Call Activity.SetStatus(ActivityStatusCode, string?) instead. This method will be removed in a future version.")]
```

Reply:
Do we _really_ need this? _Really really_? 🤣 Because there are 3 other obsoletes in this file we should update too if we do. And probably all the others ones sprinkled around the code. But IMO we don't _really_ need it. The existing message is good enough IMO, and there is also a doc link "see" ref thing in `<remarks>` which will jump users to the overload.

Reply:
> @reyang, If you prefer, we could use this.

I don't prefer any of these, I like the current one.

Reply:
Up to you : )

The reason behind this - the name is of the method is not changing. Only set of parameters. For RecordException there is a recommendation to call AddException.

Feel free to close the PR if you think that it is not worth to merge.

Reply:
Maybe send users to this URL https://learn.microsoft.com/dotnet/api/system.diagnostics.activity.setstatus from the warning message.